src/compas/files/ply.py

```diff
@@ -137,154 +146,149 @@ def read(self):
-
-                if line == 'end_header':
+                        self.face_properties.append(
+                            (property_name, property_type))
+                else:
                     element_type = None
-                    self.end_header = file.tell()
-                    break
+                    raise Exception(
+                        'Not a valid PLY file. Malformed property line.')
+                continue
+
+            if line == 'end_header':
+                element_type = None
+                self.end_header = line_number
+                break
 
     # ==========================================================================
     # read the data
     # ==========================================================================
 
     def read_data(self):
         if not self.end_header:
-            raise Exception('header has not been read, or the file is not valid')
-        with open(self.filepath) as self.file:
-            self.file.seek(self.end_header)
-            for section in self.sections:
-                if section == 'vertex':
-                    self.read_vertices()
-                elif section == 'edge':
-                    self.read_edges()
-                elif section == 'face':
-                    self.read_faces()
-                else:
-                    print('user-defined elements are not supported: {0}'.format(section))
-                    pass
+            raise Exception(
+                'Header has not been read, or the file is not valid')
+
+        for section in self.sections:
+            if section == 'vertex':
+                self.read_vertices()
+            elif section == 'edge':
+                self.read_edges()
+            elif section == 'face':
+                self.read_faces()
+            else:
+                raise Exception('Not a valid PLY file.')
 
     def read_data_binary(self):
         if not self.end_header:
-            raise Exception('header has not been read, or the file is not valid')
-        with open(self.filepath, 'rb') as self.file:
-            self.file.seek(self.end_header)
+            raise Exception(
+                'Header has not been read, or the file is not valid')
+        with open(self.location, 'rb') as self.file:
+            self.file.seek(self.end_header + 1)
             for section in self.sections:
                 if section == 'vertex':
-                    self.read_vertices_binary_wo_numpy()
+                    self.read_vertices_binary()
                 elif section == 'edge':
-                    self.read_edges_binary_wo_numpy()
+                    self.read_edges_binary()
                 elif section == 'face':
-                    self.read_faces_binary_wo_numpy()
+                    self.read_faces_binary()
                 else:
-                    print('user-defined elements are not supported: {0}'.format(section))
-                    pass
+                    raise Exception(
```

Comment:
is the entire process supposed to fail because the file contains some non-standard elements as well?

Reply:
It did before, but I think this should be changed to just ignore user-defined elements.

Reply:
+1

Reply:
Changed. Should probably set up a test file to make sure it passes.